pilot/pkg/features/pilot.go

```diff
@@ -174,11 +174,12 @@ var (
 			"Gateways with same selectors in different namespaces will not be applicable.",
 	).Get()
 
-	InboundProtocolDetectionTimeout = env.RegisterDurationVar(
+	// nolint
+	InboundProtocolDetectionTimeout, InboundProtocolDetectionTimeoutSet = env.RegisterDurationVar(
```

Comment:
Previously, even if you set meshconfig protocol detection timeout, this always overrode the option. Instead, we will only override if its explicitly set by the user. This allows us, and the user, to avoid configuring two places